Fixed 404 page

kabugatti opened the pull request:
<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**

This is a feature request for improving the 404 error-handler page that was very generic initially. The updated page explains why the user encountered that error, offers a solution by checking if there's a typo, and finally offers a button to go back to the home page.

**Issue Number:**

Fixes #388 

**Did you add tests for your changes?**

No

**Snapshots/Videos:**

<!--Add snapshots or videos wherever possible.-->
![Solution](https://res.cloudinary.com/dv765kdgq/image/upload/v1678553304/solution_mkmyf0.png)
**If relevant, did you update the documentation?**
 I did not update the documentation directly, just the 404 page.

<!--Add link to Talawa-Docs.-->

**Summary**
I noticed that if a user typed a wrong URL, they wasn't enough information to offer proper guidance. That's why I wanted to add a better 404 page that would be more user-centered and professional.

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->

**Does this PR introduce a breaking change?**
No
<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-docs/blob/develop/CONTRIBUTING.md)?**
Yes
<!--Yes or No-->

kabugatti committed:
[feat]: Swizzled NotFound Component

kabugatti committed:
[chore]: Update error page contents

kabugatti committed:
[styles]: Add 404 page styles

kabugatti committed:
[feature]: Add 404.png to errors folder in img folder

kabugatti committed:
[chore]: Update dependencies

kabugatti committed:
[styles]: Ruduce button font-size

Cioppolo14 reviewed (changes requested):
@grand-rick001 Could you please update your docusaurus? That will remove the need for the yarn.lock file.  Thank you!

kabugatti commented:
Okay, noted

kabugatti commented:
> @grand-rick001 Could you please update your docusaurus? That will remove the need for the yarn.lock file. Thank you!

Do you mean updating it to version 2 or ?

kabugatti committed:
[docs]: Update docusaurus

salman-farooqi commented on src/theme/NotFound.js:
```diff
@@ -0,0 +1,51 @@
+                  <Translate
+                    id="theme.NotFound.button"
+                    description="Button on the 404 page that redirects to the Homepage">
+                    HOME PAGE
```
(It's my opinion, I maybe am wrong)
What if you use the 'Homepage' icon instead of text?
If it's not allowed by @palisadoes, it would look more good if you update the button's style/color and also use hover selector.

grand-rick001 replied:
Okay, I'll try it out

kabugatti replied:
Did some styling on the HOME button, the icon wouldn't be more user friendly in my opinion @salman-farooqi

kabugatti committed:
Merge branch 'PalisadoesFoundation:develop' into my-dev

kabugatti committed:
[style]: Add style for the HOME button

kabugatti committed:
[chore]: Edit Home Button content

kabugatti committed:
Merge branch 'my-dev' of github.com:grand-rick001/talawa-docs into my-dev

kabugatti commented:
I've finished up on setting the 404 page, updated the Docusaurus as needed and added styling to the Home button. Now just waiting for a review to finalize

kabugatti committed:
[chore]: Remove image before HOME button

kabugatti commented:
@palisadoes please review this PR

kabugatti committed:
[chore]: Update yarn

kabugatti committed:
[chore]: Modify yarn.lock

kabugatti committed:
[chore]: Add .yarnrc.yml

kabugatti committed:
git branchMerge branch 'my-dev' of github.com:grand-rick001/talawa-docs into my-dev

kabugatti committed:
[docs]: Updated yarn.lock

kabugatti committed:
[delete]: Delete additional yarn files

Cioppolo14 commented:
@grand-rick001 Docusaurus latest version 2.3.1, please use the yarn.lock from Talawa-Docs repository.

kabugatti committed:
Merge branch 'PalisadoesFoundation:develop' into my-dev

Cioppolo14 commented:
@grand-rick001 Could you please address the conflicting files? Yarn.lock has recently been updated.  Thank you!

Cioppolo14 commented:
@grand-rick001 I think it might be best to close this request, ensure you have the latest Docusaurus and yarn.lock and resubmit. I see over 1800 files that have changes right now.

kabugatti commented:
Well, I used the `yarn` command to update docusaurus from the `yarn.lock` file as requested @Cioppolo14

kabugatti commented:
I'll do a quick solution then request for a review

kabugatti commented:
How's it now @Cioppolo14 ?

Cioppolo14 commented:
@grand-rick001 Its much better, but there is still yarn.lock differences.  Can you fix that?

kabugatti commented:
> @grand-rick001 Its much better, but there is still yarn.lock differences. Can you fix that?

Alright, on it, so I have to copy the `yarn.lock` file on my PR? @Cioppolo14

Cioppolo14 commented:
> > @grand-rick001 Its much better, but there is still yarn.lock differences. Can you fix that?
> 
> Alright, on it, so I have to copy the `yarn.lock` file on my PR? @Cioppolo14

Copy the yarn.lock file from the repository and try that.  We want there to be no differences in your yarn.lock from the main Talawa-Docs repository, so that when we merge, no changes are made to the yarn.lock.

kabugatti committed:
[feat(update)]: Update yarn.lock

kabugatti commented:
Yap, done @Cioppolo14

Cioppolo14 commented:
@grand-rick001 Could you not submit the yarn.lock file? Then we don't have to worry about changing it.

kabugatti committed:
[chore]: Add yarn.lock to .gitignore

kabugatti commented:
Done @Cioppolo14

Cioppolo14 commented:
@grand-rick001 Let's not use gitignore to work around the issue.  Let's get yarn.lock out of the files changed section.

Cioppolo14 commented:
@grand-rick001 We have a limit on the testing resources we use on GitHub. We only have 2000 minutes per month as an open source community. By making numerous commits you reduce the chances of others being able to contribute to the code base in the same way you are now.
Please read our documentation on how to test your code on your local system. Additional commits should be the result of suggestions from the PR’s reviewers.

Please stop so that everyone can get a fair chance.

kabugatti commented:
Okay 👍

kabugatti commented:
@Cioppolo14 I would really like to finish this pull request, can we please discuss the details. I've messaged you on slack.

kabugatti committed:
[update]: Update docusaurus

kabugatti committed:
Merge branch 'PalisadoesFoundation:develop' into my-dev

kabugatti committed:
[chore]: Removed yarn.lock from .gitignore

kabugatti committed:
Merge branch 'my-dev' of github.com:grand-rick001/talawa-docs into my-dev

kabugatti commented:
I noticed that I wasn't looking at the files changed to check the differences, ha, my bad. Now I know!

Cioppolo14 commented:
We have a new website design coming, so I'm going to close this PR and related issue and let the new design address it.
